docs: update upgrade guide with `talosctl upgrade-k8s` (#2526)
talos-systems/talos: docs/website/content/v0.6/en/guides/upgrading/index.md

If you don't already have one, you can get one by running:

```bash
talosctl --nodes <master node> kubeconfig
```

### Automated Kubernetes Upgrade

> New command `talosctl upgrade-k8s` can be used to automate the Kubernetes upgrade process, this command was first released with Talos 0.6.1.

Reviewer: ```suggestion
In Talos v0.6.1 we introduced the `upgrade-k8s` command in `talosctl`.
This command can be used to automate the Kubernetes upgrade process.
For example, to upgrade from Kubernetes v1.18.6 to v1.19.0 run:
```

Author: thanks!

```bash
$ talosctl --nodes <master node> upgrade-k8s --from 1.18.6 --to 1.19.0
updating pod-checkpointer grace period to "0m"

> sleeping 5m0s to let the pod-checkpointer self-scheckpoint be updated

Reviewer 2: Not sure if this is a typo in the actual upgrade code, but `self-scheckpoint`

> sleeping 5m0s to let the pod-checkpointer self-scheckpoint be updated

Reviewer 2: ```suggestion
sleeping 5m0s to let the pod-checkpointer self-checkpoint be updated
```

Author: :facepalm: that's in the code, I'll fix it as well

temporarily taking "kube-apiserver" out of pod-checkpointer control
updating daemonset "kube-apiserver" to version "1.19.0"
updating daemonset "kube-controller-manager" to version "1.19.0"
updating daemonset "kube-scheduler" to version "1.19.0"
updating daemonset "kube-proxy" to version "1.19.0"
updating pod-checkpointer grace period to "5m0s"
```

### Manual Kubernetes Upgrade

> Kubernetes can be upgraded manually as well by following the steps outlined below, they are equivalent to the steps performed by the `talosctl upgrade-k8s` command.

Reviewer 2: ```suggestion
Kubernetes can be upgraded manually as well by following the steps outlined below.
They are equivalent to the steps performed by the `talosctl upgrade-k8s` command.
```

Author: :+1:

#### pod-checkpointer

> Talos runs `pod-checkpointer` component which helps to recover control plane components (specifically, API server) if control plane is not healthy, but the way
> checkpoints interact with API server upgrade might make upgrade take a lot of time due to race condition in API server listen port.

Reviewer 2: ```suggestion
Talos runs `pod-checkpointer` component which helps to recover control plane components (specifically, API server) if control plane is not healthy.
However, the way checkpoints interact with API server upgrade may make an upgrade take a lot longer due to a race condition in API server listen port.
```

Author: :heart:
